extract-method-plugin/src/main/kotlin/org/jetbrains/research/extractMethod/BaseRunner.kt

```diff
@@ -0,0 +1,46 @@
+package org.jetbrains.research.extractMethod
+
+import org.apache.logging.log4j.LogManager
+import org.jetbrains.research.extractMethod.core.extractors.NegativeRefactoringsExtractionRunner
+import org.jetbrains.research.extractMethod.core.extractors.PositiveRefactoringsExtractionRunner
+import org.jetbrains.research.pluginUtilities.openRepository.getKotlinJavaRepositoryOpener
+import org.jetbrains.research.pluginUtilities.preprocessing.getKotlinJavaPreprocessorManager
+import java.io.FileWriter
+import java.nio.file.Files
+import java.nio.file.Path
+import kotlin.streams.toList
+
+private val preprocessor = getKotlinJavaPreprocessorManager(null)
+private val repositoryOpener = getKotlinJavaRepositoryOpener()
+
+class BaseRunner {
+    private val logger = LogManager.getLogger(BaseRunner::class)
+
+    private fun getSubdirectories(path: Path): List<Path> {
+        return Files.walk(path, 1)
+            .filter { Files.isDirectory(it) && !it.equals(path) }
+            .toList()
+    }
+
+    fun runPositives(inputDir : Path, fw : FileWriter) {
+        val runner = PositiveRefactoringsExtractionRunner(fw);

```

Comment:
what about extracting base interface named `RefactoringExtractor` and add two implementations -- one for `PositiveRefactoringsExtractor` and another one for `NegativeRefactoringsExtractor`? 
they both have methods with the same names.

Reply:
Was about to do it today, actually!